wandb/sdk/artifacts/artifact.py

```diff
@@ -1663,6 +1663,33 @@ def download(
         root: Optional[str] = None,
         recursive: bool = False,
         allow_missing_references: bool = False,
+    ) -> FilePathStr:
+        self._ensure_logged("download")
+
+        root = root or self._default_root()
+        self._add_download_root(root)
```

Comment:
Nit: this is done again in `_download`. Decide whether this should be done in the foreground or background process.